x-pack/plugins/monitoring/public/alerts/lib/get_alert_panels_by_category.tsx

```diff
@@ -10,73 +10,89 @@ import { EuiText, EuiToolTip } from '@elastic/eui';
 import { AlertPanel } from '../panel';
 import { ALERT_PANEL_MENU } from '../../../common/constants';
 import { getDateFromNow, getCalendar } from '../../../common/formatting';
-import { IAlertsContext } from '../context';
-import { AlertState, CommonAlertStatus } from '../../../common/types/alerts';
+import {
+  AlertState,
+  CommonAlert,
+  CommonAlertState,
+  CommonAlertStatus,
+} from '../../../common/types/alerts';
 import { PanelItem } from '../types';
 import { sortByNewestAlert } from './sort_by_newest_alert';
 import { Legacy } from '../../legacy_shims';
 
```

Comment:
This view creates the nested alerts menu when clicking on the badges.  It's very confusing but I tried to clean it up a bit and add some types.  Basically it's accommodating the fact that there can be more than one type of a rule and we should show them both in the menu.

Reply:
Agreed that it's still very confusing. I would like to pull it apart more but not during this PR. For now, can we rename `alerts` on the incoming `getAlertPanelsByCategory` function, to something that is more descriptive? It's really hard to keep track of what "alerts" means at any given time. 

Also, can we add a bunch of comments for now about what these functions are doing step by step?

Reply:
I was playing around with the code to see if I could understand it better. This is where I ended up: https://gist.github.com/jasonrhodes/a6ad54b808943d46f85082df2bbfb200

I don't think we should do any of that right now, or ever necessarily. I think what you've done here is a good step for what we need right now, but if you look that over it may help you see where some comments might be useful based on how badly I've misunderstood some things, etc.